src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -100,6 +100,8 @@ public class ScriptHost : JobHost
             _proxyClient = proxyClient;
         }
 
+        public event EventHandler HostInitializing;
```

Comment:
Production testing revealed the fact that the host health check log entries I was trying to write weren't being written, because the check is happening too early in the startup path for the host.

So I'm making several changes, including this new event to allow the host manager to intercept initialization as early as possible (after config and loggers have been created, but before the bulk of initialization).